### DISCUSSION TITLE
Export `CreateObservableOptions` type so one can extend makeObservable

### DISCUSSION THREAD

**Discussion by ahoisl** [mobxjs/mobx#2716] [General]
If you want to extend or create a custom function using makeObservable, you may want to also pass in `CreateObservableOptions` to that custom function.
The `CreateObservableOptions` type is not exported though and must be imported from `dist/api/observable` or `dist/internal` - both unresolvable locations at pack time (according to dependency cruiser).

Would  be great if we can also export that type.

Suggested code change in PR #2717

**Comment by danielkcz** [comment]
Technically, you should be able to access that type with `Parameters<typeof makeObservable>[0]`, but I agree that exporting it won't do any harm.
